### PR TITLE
doc: Explain new test logging

### PR DESCRIPTION
**Pull request by maflcko**
Explain logging added in #18472 and #16975

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Just as a note, if we wanted to remove boost, `BOOST_TEST_MESSAGE` can be replaced by a simple `TEST_MESSAGE` macro with this diff:


```diff
diff --git a/src/test/main.cpp b/src/test/main.cpp
index f32243d1d3..9bccbef805 100644
--- a/src/test/main.cpp
+++ b/src/test/main.cpp
@@ -13,14 +13,19 @@
 
 #include <iostream>
 
-/** Redirect debug log to unit_test.log files */
-const std::function<void(const std::string&)> G_TEST_LOG_FUN = [](const std::string& s) {
+bool IsTestLoggingEnabled()
+{
     static const bool should_log{std::any_of(
         &boost::unit_test::framework::master_test_suite().argv[1],
         &boost::unit_test::framework::master_test_suite().argv[boost::unit_test::framework::master_test_suite().argc],
         [](const char* arg) {
             return std::string{"DEBUG_LOG_OUT"} == arg;
         })};
-    if (!should_log) return;
+    return should_log;
+}
+
+/** Redirect debug log to unit_test.cpp.log files */
+const std::function<void(const std::string&)> G_TEST_LOG_FUN = [](const std::string& s) {
+    if (!IsTestLoggingEnabled()) return;
     std::cout << s;
 };
diff --git a/src/test/util/logging.h b/src/test/util/logging.h
index 45ec44173c..f4ad232c6e 100644
--- a/src/test/util/logging.h
+++ b/src/test/util/logging.h
@@ -11,6 +11,13 @@
 #include <list>
 #include <string>
 
+bool IsTestLoggingEnabled();
+
+#define TEST_MESSAGE(blob)              \
+    if (IsTestLoggingEnabled()) {       \
+        std::cout << blob << std::endl; \
+    }
+
 class DebugLogHelper
 {
     const std::string m_message;

File: src/test/README.md [hunk 1]
```diff
@@ -17,26 +17,31 @@ and tests weren't explicitly disabled.
...
+`test_bitcoin` has some built-in command-line arguments; for
+example, to run just the `getarg_tests` verbosely:
 
-    test_bitcoin --log_level=all --run_test=getarg_tests
+    test_bitcoin --log_level=all --run_test=getarg_tests -- DEBUG_LOG_OUT
+
+`log_level` controls the verbosity of the test framework, which logs when a
+test case is entered, for example. The `DEBUG_LOG_OUT` after the two dashes
+redirects the debug log, which would normally go to a file in the test datadir
+(`BasicTestingSetup::m_path_root`), to the stdandard terminal output.
```
Comment (jonatack) [inline src/test/README.md]: standard

**Review by jonatack — COMMENTED** [review]
ACK modulo nit

File: src/test/README.md (hunk 1 above)
Comment (jonatack) [reply src/test/README.md]: or maybe it's a mix of stdout and standard :)

Comment (MarcoFalke) [reply src/test/README.md]: > or maybe it's a mix of stdout and standard :)

Yes, in my brain I wanted to write `stdout`, but then decided to write the longer form after I started :)

Comment (MarcoFalke) [reply src/test/README.md]: Fixed in commit 77777

**Review by jonatack — COMMENTED** [review]
ACK 7777703

**Pull request merged**